src/index.js

```diff
@@ -25,40 +25,52 @@ const upcomingTripTemplate = document.querySelector('.future-trips-template')
 
 const pageLoad = () => {
   let rando = (Math.ceil(Math.random() * 49))
-  getTraveler(rando)
-  .then(traveler => traveler = new Traveler(traveler.id, traveler.name, traveler.travelerType))
-  .then(traveler => dashboardGreeting.innerText = "Hello, " + (traveler.returnFirstNameLastInitial()))
 
-  getTrips()
-    .then(trips => trips = trips.map(trip => { 
-    return new Trip(trip)}))
-    .then(trips => trips.filter(trip => {
+  const travelerResults = getTraveler(rando)
+    .then(traveler => traveler = new Traveler(traveler.id, traveler.name, traveler.travelerType))
+    .then(traveler => dashboardGreeting.innerText = "Hello, " + (traveler.returnFirstNameLastInitial()))
+
+  const tripsResults = getTrips()
+
+  const placeResults = getDestinations()
+    .then(destinations => destinations = destinations.map(destination => {
+      return new Destination(destination)
+    }))
+    
+
+  Promise.all([travelerResults, tripsResults, placeResults])
+    .then(values => values[1].map(trip => {
+      let placeName = values[2].find(place => {
+        if (place.id === trip.destinationID) {
+          return place.name
```

Comment:
- `.find()` cannot change what it is returning;
  - it must return just the value you're initially looking to find
- I'd have this find the place, then have another var check this destination and save the name

Reply:
Perfect! I have changed that so that going forward it is just returning the whole destination object and passing it into the trip. This should make calculating the cost much easier.